internal/action/handle.go

```diff
@@ -49,50 +50,62 @@ func ShowCurrent() {
 }
 
 func SetCurrent(source *registry.RegistryDataSource, args []string) int {
-
-	name := "npm"
-
-	if len(args) >= 1 {
-		name = args[0]
-	}
-	uri, ok := source.Registry[name]
-	if !ok {
-		logger.Error(internal.StringJoin("[Grm]: Can't found alias", name, "in your .nrmrc file. Please check it exist.", registry.Eol()))
+	defer func() {
+		if err := recover(); err != nil {
+			logger.Warn("[Grm]: Plese pass an alias.")
+			return
+		}
+	}()
+	name := internal.PickArgs(args, 0)
+	uri, err := getRegistryMeta(name, source.Registry, func(n string) (string, error) {
+		return "", errors.New(internal.StringJoin("[Grm]: Can't found alias", name, "in your .nrmrc file. Please check it exist."))
+	})
+	if err != nil {
+		logger.Error(internal.StringJoin(err.Error(), registry.Eol()))
 		return 1
 	}
-	err := registry.WriteNpm(uri)
+	err = registry.WriteNpm(uri)
 	if err != nil {
 		logger.Error(internal.StringJoin("[Grm]: error with", err.Error(), registry.Eol()))
 		return 1
 	}
 	logger.Success(internal.StringJoin("[Grm]: use", name, "success~", registry.Eol()))
 	return 0
+
 }
 
 // del .nrm file registry alias
 
 func DelRegistry(source *registry.RegistryDataSource, args []string) int {
-
-	if len(args) == 0 {
-		return 0
-	}
-	name := args[0]
-
-	_, ok := source.UserRegistry[name]
-
-	if !ok {
-		logger.Error(internal.StringJoin("[Grm]: Can't found alias", name, "in your .nrmrc file. Please check it exist.", registry.Eol()))
+	defer func() {
+		if err := recover(); err != nil {
+			logger.Warn("[Grm]: Plese pass an alias.")
```

Comment:
If a panic occurs, we should log the panic information so that we can locate the problem code.

Reply:
> If a panic occurs, we should log the panic information so that we can locate the problem code.

Yes, But in production mode. shall we need it? From the `PickArgs` function we can get the panic message is `Invalid prt` Is it valid information for the user?

Reply:
I will think about.

Reply:
> I will think about.

Yes,  just need to log once. My fault.